Firestore/core/src/firebase/firestore/remote/serializer.cc

```diff
@@ -475,7 +472,7 @@ FieldValue DecodeNestedFieldValue(pb_istream_t* stream) {
  * @param kv The individual key/value pair to write.
  */
 void EncodeFieldsEntry(Writer* writer,
-                       const std::pair<std::string, FieldValue>& kv) {
+                       const FieldValue::ObjectT::value_type& kv) {
```

Comment:
Could we define alias for this as well instead of refer directly? Later we may change ObjectT to use immutable container, which may not define alias type with exact same name `value_type`.

Reply:
I'd vote for adding `value_type` to the immutable container, it's common STL convention.

Reply:
SGTM

Reply:
Yeah, Costa's suggestion is my preference too. Worst case scenario: we switch over to an immutable container which doesn't define it, and we'll get a nice big compiler error. At that point we can either (a) add value_type to the container (my preference) or (b) add another alias to FieldValue.